src/kboot.c

```diff
@@ -47,7 +47,9 @@ static char *chosen_params[MAX_CHOSEN_PARAMS][2];
 
 extern const char *const m1n1_version;
 
+#ifdef DO_GPU_INIT
 int dt_set_gpu(void *dt);
```

Comment:
I'd probably do something like 
```suggestion
#ifdef DP_GPU_INIT
int dt_set_gpu(void *dt);
#else
static int dt_set_gpu(void *dt)
{
    printf("Cannot setup GPU data in devicetree because m1n1 was built without rust support");
}
#endif
```
here just to be as verbose as possible